src/rnp/fficli.cpp

```diff
@@ -1347,109 +1347,59 @@ cli_rnp_keys_matching_strings(cli_rnp_t *                     rnp,
  *
  *  @return true if path constructed successfully, or false otherwise
  **/
-static bool
-rnp_path_compose(
-  const char *dir, const char *subdir, const char *filename, char *res, size_t res_size)
+static std::string
+rnp_path_compose(const std::string &dir,
+                 const std::string &subdir,
+                 const std::string &filename)
 {
-    int pos;
-
-    /* checking input parameters for conrrectness */
-    if (!dir || !filename || !res) {
-        return false;
-    }
-
-    /* concatenating dir, subdir and filename */
-    if (strlen(dir) > res_size - 1) {
-        return false;
-    }
-
-    strcpy(res, dir);
-    pos = strlen(dir);
-
-    if (subdir) {
-        if ((pos > 0) && (res[pos - 1] != '/')) {
-            res[pos++] = '/';
+    std::string res = dir;
+    if (!subdir.empty()) {
+        if (!res.empty() && (res.back() != '/')) {
```

Comment:
should we support backslash on Windows?

Reply:
@rrrooommmaaa Most likely we should, but in another PR, since this one is aimed to remove `strcpy()` usage.

Reply:
Ok, I added #1290